### PR TITLE
Fix DMs in sidebar

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
This PR fix some inconsistencies with the Browser way of showing DMs:

- Archived channels not properly being hidden
- Opened channels without messages not properly being shown
- Unavailable users on the database not being fetched from the server (and therefore, not being filtered out if needed)

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-49935
Fix https://mattermost.atlassian.net/browse/MM-53024
Fix https://mattermost.atlassian.net/browse/MM-51530

#### Release Note
```release-note
Remove some inconsistencies in the way DMs are shown in the sidebar between Mobile and Browser.
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]
lgtm

File: app/actions/remote/preference.ts [hunk 1]
```diff
@@ -128,6 +134,87 @@ export const deleteSavedPost = async (serverUrl: string, postId: string) => {
...
+                    name,
+                    value: 'true',
+                },
+                {
+                    user_id: userId,
+                    category: Preferences.CATEGORIES.CHANNEL_OPEN_TIME,
+                    name: channel.id,
+                    value: Date.now().toString(),
+                },
+            );
```
Comment (enahum) [inline app/actions/remote/preference.ts:208]: this is almost identical to `openChannelIfNeeded` perhaps create a function that you can call from both actions?

**Review by harshilsharma63 — APPROVED** [review]
Looks good!

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes. I've tested the 3 tickets. The user list is same on web app and mobile

**Pull request merged**